### PR TITLE
trove: change mysql backend to postgres

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

**Comment by dirkmueller** [comment]
[Switch trove to use the main postgresql database](https://trello.com/c/yDU1cEG9/2159-switch-trove-to-use-the-main-postgresql-database)

**Comment by Itxaka** [comment]
Wont work until https://build.opensuse.org/request/show/448765 is merged and available.

**Comment by Itxaka** [comment]
Fixing it manually for the missing trove-manage (which is on the openstack-trove-api package instead of on openstack-trove) seems to make it work properly:

```
root@d52-54-77-77-77-01:~ # trove create test manila-service-image-flavor --datastore mysql --datastore_version 5.5 --size 5
+-------------------+--------------------------------------+
| Property          | Value                                |
+-------------------+--------------------------------------+
| created           | 2017-01-04T17:02:26                  |
| datastore         | mysql                                |
| datastore_version | 5.5                                  |
| flavor            | 100                                  |
| id                | 5d667231-7562-43cc-a0ee-19b798ec8dac |
| name              | test                                 |
| status            | BUILD                                |
| updated           | 2017-01-04T17:02:26                  |
| volume            | 5                                    |
+-------------------+--------------------------------------+

root@d52-54-77-77-77-01:~ # su - postgres
postgres@d52-54-77-77-77-01:~ > psql
psql (9.4.9)
Type "help" for help.

postgres=# \connect trove;
You are now connected to database "trove" as user "postgres".
trove=# \dt
                      List of relations
 Schema |                 Name                 | Type  | Owner 
--------+--------------------------------------+-------+-------
 public | agent_heartbeats                     | table | trove
 public | backups                              | table | trove
 public | capabilities                         | table | trove
 public | capability_overrides                 | table | trove
 public | clusters                             | table | trove
 public | conductor_lastseen                   | table | trove
 public | configuration_parameters             | table | trove
 public | configurations                       | table | trove
 public | datastore_configuration_parameters   | table | trove
 public | datastore_version_metadata           | table | trove
 public | datastore_versions                   | table | trove
 public | datastores                           | table | trove
 public | dns_records                          | table | trove
 public | instance_faults                      | table | trove
 public | instance_modules                     | table | trove
 public | instances                            | table | trove
 public | migrate_version                      | table | trove
 public | modules                              | table | trove
 public | quota_usages                         | table | trove
 public | quotas                               | table | trove
 public | reservations                         | table | trove
 public | root_enabled_history                 | table | trove
 public | security_group_instance_associations | table | trove
 public | security_group_rules                 | table | trove
 public | security_groups                      | table | trove
 public | service_images                       | table | trove
 public | service_statuses                     | table | trove
 public | usage_events                         | table | trove
(28 rows)


```

File: chef/cookbooks/trove/libraries/helpers.rb [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
-    # running on the same node
-    "mysql://#{node[:trove][:db][:user]}:"\
-    "#{db_password}@127.0.0.1/"\
-    "#{node[:trove][:db][:database]}"
+    # get Database data
+    db_settings = CrowbarOpenStackHelper.database_settings(node, "trove")
+    sql_connection = "#{db_settings[:url_scheme]}://#{node[:trove][:db][:user]}:"\
+                 "#{node[:trove][:db][:password]}@#{db_settings[:address]}/"\
+                 "#{node[:trove][:db][:database]}"
+    sql_connection
```
Comment (nicolasbock) [inline chef/cookbooks/trove/libraries/helpers.rb]: Is that line necessary? I am no Ruby expert, but I though that a function's return value is the result of the last command.

Comment (aplanas) [inline chef/cookbooks/trove/libraries/helpers.rb]: I agree with @nicolasbock, you can return the string like in the original code

Comment (Itxaka) [inline chef/cookbooks/trove/libraries/helpers.rb]: Indeed. I keep forgetting the weird ruby way of returning from functions

**Pull request merged**